### PR TITLE
Refactor copy packages command to support regional ECR

### PR DESCRIPTION
**Pull request by d8660091**
*Issue #, if available:*

*Description of changes:* 
- no more dependency on running docker containers
- support for regional curated packages

*Testing (if applicable):*
- Manually tested copy packages from regional ECR to local registry
- Manually tested copy packages from global ECRs to local registry
- Manually tested copy packages from global ECRs to regional ECR

*Documentation added/planned (if applicable):*

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: docs/content/en/docs/getting-started/airgapped/airgap-steps.md [hunk 1]
```diff
@@ -31,14 +31,18 @@ toc_hide: true
...
-
-   ```bash
-   eksctl anywhere copy packages \
-     --bundle ${BUNDLE_RELEASE_YAML_PATH} \
-     --dst-cert ${REGISTRY_MIRROR_CERT_PATH} \
-     ${REGISTRY_MIRROR_URL}
-   ```
+      If your EKSA cluster is running in an airgapped environment and you set up a local registry mirror, you can copy curated packages from Amazon ECR to your local registry mirror with the following command. 
+      
+      `$KUBEVERSION` must equals to the `spec.kubernetesVersion` of your EKS Anywhere cluster specification.
```
Comment (chrisnegus) [inline docs/content/en/docs/getting-started/airgapped/airgap-steps.md:37]:
```suggestion
      `$KUBEVERSION` be must equal to the `spec.kubernetesVersion` of your EKS Anywhere cluster specification.
```

File: docs/content/en/docs/reference/eksctl/anywhere_copy.md [hunk 2]
```diff
@@ -26,5 +26,5 @@ Copy EKS Anywhere resources and artifacts
 ### SEE ALSO
 
 * [anywhere](../anywhere/)	 - Amazon EKS Anywhere
-* [anywhere copy packages](../anywhere_copy_packages/)	 - Copy curated package images and charts from a source to a destination
+* [anywhere copy packages](../anywhere_copy_packages/)	 - Copy curated package images and charts from source regisries to a destination registry
```
Comment (chrisnegus) [inline docs/content/en/docs/reference/eksctl/anywhere_copy.md:29]:
```suggestion
* [anywhere copy packages](../anywhere_copy_packages/)	 - Copy curated package images and charts from source registries to a destination registry
```

File: docs/content/en/docs/reference/eksctl/anywhere_copy_packages.md [hunk 3]
```diff
@@ -5,11 +5,11 @@ linkTitle: "anywhere copy packages"
 
 ## anywhere copy packages
 
-Copy curated package images and charts from a source to a destination
+Copy curated package images and charts from source regisries to a destination registry
```
Comment (chrisnegus) [inline docs/content/en/docs/reference/eksctl/anywhere_copy_packages.md:8]:
```suggestion
Copy curated package images and charts from source registries to a destination registry
```

File: docs/content/en/docs/reference/eksctl/anywhere_copy_packages.md [hunk 4]
```diff
@@ -18,13 +18,13 @@ anywhere copy packages <destination-registry> [flags]
...
 
 ```
-      --aws-region string   Region to copy images from
-  -b, --bundle string       EKS-A bundle file to read artifact dependencies from
-      --dry-run             Dry run copy to print images that would be copied
-      --dst-cert string     TLS certificate for destination registry
-  -h, --help                help for packages
-      --insecure            Skip TLS verification while copying images and charts
-      --src-cert string     TLS certificate for source registry
+      --dry-run                     Dry run will not really copy the artifacts, but shows what artifacts would be copied
```
Comment (chrisnegus) [inline docs/content/en/docs/reference/eksctl/anywhere_copy_packages.md:21]:
```suggestion
      --dry-run                     Dry run will show what artifacts would be copied, but not actually copy them
```

File: docs/content/en/docs/reference/eksctl/anywhere_copy_packages.md [hunk 5]
```diff
@@ -18,13 +18,13 @@ anywhere copy packages <destination-registry> [flags]
...
+      --dst-insecure                Skip TLS verification against the destination registry
+      --dst-plain-http              Whether to use plain http for destination registry
```
Comment (chrisnegus) [inline docs/content/en/docs/reference/eksctl/anywhere_copy_packages.md:23]:
```suggestion
      --dst-plain-http              Whether or not to use plain http for destination registry
```
Is the alternative https?

File: docs/content/en/docs/reference/eksctl/anywhere_copy_packages.md [hunk 6]
```diff
@@ -18,13 +18,13 @@ anywhere copy packages <destination-registry> [flags]
...
+  -h, --help                        help for packages
+      --kube-version string         The kube version of the package bundle to copy
```
Comment (chrisnegus) [inline docs/content/en/docs/reference/eksctl/anywhere_copy_packages.md:25]:
```suggestion
      --kube-version string         The Kubernetes version of the package bundle to copy
```

**Review by chrisnegus — COMMENTED** [review]
I left a few comments.

File: cmd/eksctl-anywhere/cmd/copypackages.go [hunk 7]
```diff
@@ -1,201 +1,120 @@
...
-	}
-	return nil
+	// copy package bundle yaml after charts and images
+	tag := getPackageBundleTag(cpc.kubeVersion)
+	_, err = orasCopy(ctx, curatedpackages.ImageRepositoryName, cpc.srcChartRegistry, tag, cpc.destRegistry, tag)
+	return err
 }
 
-// Since package bundle doesn't contain tags, we get tags from chart values.
 func getTagsFromCharts(ctx context.Context, helm *executables.Helm, charts []registry.Artifact) (map[string]string, error) {
```
[the remark this answers is not in the document]
Comment (d8660091) [reply cmd/eksctl-anywhere/cmd/copypackages.go]: Good catch, removed

**Pull request merged**